packages/framework-provider-local-infrastructure/test/scheduler.test.ts

```diff
@@ -0,0 +1,84 @@
+import { BoosterConfig } from '@boostercloud/framework-types'
+import { restore, SinonStub, SinonStubbedInstance, createStubInstance, replace, stub, useFakeTimers } from 'sinon'
+import { configureScheduler, createCronExpression, buildScheduledCommandInfo } from '../src/scheduler'
+import { expect } from './expect'
+import { describe } from 'mocha'
+import { random } from 'faker'
+
+declare class UserProject {
+  constructor()
+  boosterTriggerScheduledCommand: (rawRequest: unknown) => void
+}
+describe('Local Scheduler', () => {
+  let mockUserProjectStub: SinonStubbedInstance<UserProject>
+  let mockScheduledCommand: any
+  let mockConfig: BoosterConfig
+  let mockScheduledCommandName: string
+  let queryStub: SinonStub
+  let clock: any
+
+  beforeEach(() => {
+    class CheckCart {
+      public static async handle(): Promise<void> {
+        console.log('handle')
+      }
+    }
+    class UserProject {
+      boosterTriggerScheduledCommand(rawRequest: unknown): void {
+        console.log('rawRequest: ', rawRequest)
+      }
+    }
+
+    clock = useFakeTimers()
+
+    mockScheduledCommandName = random.word()
+    queryStub = stub()
+
+    mockUserProjectStub = createStubInstance(UserProject)
+    replace(mockUserProjectStub, 'boosterTriggerScheduledCommand', queryStub as any)
+
+    mockConfig = buildConfig()
+    mockConfig.scheduledCommandHandlers[mockScheduledCommandName] = {
+      class: CheckCart,
+      scheduledOn: {},
+    }
+
+    mockScheduledCommand = buildScheduledCommandInfo(mockConfig, mockScheduledCommandName)
+  })
+
+  afterEach(() => {
+    restore()
+  })
+
+  describe('buildScheduledCommandInfo', () => {
+    it('should return expected scheduled command info', async () => {
+      const result = await buildScheduledCommandInfo(mockConfig, mockScheduledCommandName)
+      expect(result).to.be.deep.equal({
+        name: mockScheduledCommandName,
+        metadata: mockConfig.scheduledCommandHandlers[mockScheduledCommandName],
+      })
+    })
+  })
+
+  describe('createCronExpression', () => {
+    it('should return expected cron expression', async () => {
+      const result = await createCronExpression(mockScheduledCommand.metadata)
+      expect(result).to.be.equal('* * * * * *')
+    })
+  })
+
+  describe('configureScheduler', () => {
+    it('should call scedule job', async () => {
+      const results = configureScheduler(mockConfig, mockUserProjectStub)
+      clock.tick(3250)
+      setTimeout(() => {
+        results.forEach((result) => result.cancel())
+      }, 3250)
+      expect(queryStub).to.have.callCount(3)
+    })
+  })
```

Comment:
One thing you could do to make this test faster and avoid returning the job instances is to mock the `node-schedule` library and check that the schedule method is called with the right parameters. 
If you also want to verify that the scheduled function is triggering the `boosterTriggerScheduledCommand` function you can provide a fake that runs the function passed to the scheduler right away:

```typescript
import * as scheduler from 'node-schedule'
...
  describe('configureScheduler', () => {
    it('should call scedule job', async () => {
      replace(
      const fakeScheduleJob = fake((name, cronExpression, scheduledFunction) => scheduledFunction())
      replace(scheduler, 'scheduleJob', fakeScheduledJob)
      replace(UserProject, 'boosterTriggerScheduledCommand', fake())

      configureScheduler(mockConfig, mockUserProjectStub)

      expect(scheduler.scheduleJob).to.have.been.calledWith(/* -- the right params */)
      expect(UserProject.boosterTriggerScheduledCommand).to.have.been.calledWith(/* ... */)
    })
  })
```